vertical-pod-autoscaler/pkg/recommender/logic/estimator.go

```diff
@@ -126,18 +135,24 @@ func (e *confidenceMultiplier) GetResourceEstimation(s *model.AggregateContainer
 	return scaledResources
 }
 
-// Returns resources computed by the underlying estimator with the additional
-// "safety margin" applied. Each resource is transformed as follows:
-//     resource = originalResource + max(originalResource * marginFraction, minMargin).
-func (e *safetyMargin) GetResourceEstimation(s *model.AggregateContainerState) model.Resources {
+func (e *marginEstimator) GetResourceEstimation(s *model.AggregateContainerState) model.Resources {
 	originalResources := e.baseEstimator.GetResourceEstimation(s)
 	newResources := make(model.Resources)
 	for resource, resourceAmount := range originalResources {
 		margin := model.ScaleResource(resourceAmount, e.marginFraction)
-		if margin < e.minMargin[resource] {
-			margin = e.minMargin[resource]
-		}
 		newResources[resource] = originalResources[resource] + margin
 	}
 	return newResources
 }
+
+func (e *minResourcesEstimator) GetResourceEstimation(s *model.AggregateContainerState) model.Resources {
+	originalResources := e.baseEstimator.GetResourceEstimation(s)
+	newResources := make(model.Resources)
+	for resource, resourceAmount := range originalResources {
+		if resourceAmount < e.minResources[resource] {
```

Comment:
why e.maxResources is not included?

Reply:
This is minimum resources recommended to a pod. Recommendation has no max boundary (not to confuse with min/maxAllowed that is part of resourcePolicy).